Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Booking.hs

```diff
@@ -107,9 +117,16 @@ findFareForCancelledBookings :: MonadFlow m => [Id Booking] -> m Money
 findFareForCancelledBookings bookingIds = findAllWithKV [Se.And [Se.Is BeamB.status $ Se.Eq CANCELLED, Se.Is BeamB.id $ Se.In $ getId <$> bookingIds]] <&> sum . map Domain.Types.Booking.estimatedFare
 
 instance FromTType' BeamB.Booking Booking where
+  fromTType' :: MonadFlow m => BeamB.Booking -> m (Maybe Booking)
   fromTType' BeamB.BookingT {..} = do
-    fl <- QBBL.findById (Id fromLocationId) >>= fromMaybeM (InternalError $ "FromLocation not found in booking for fromLocationId: " <> show fromLocationId)
-    tl <- QBBL.findById (Id toLocationId) >>= fromMaybeM (InternalError $ "ToLocation not found in booking for toLocationId: " <> show toLocationId)
+    mappings <- QLM.findByEntityId id
+    let fromLocationMapping = filter (\loc -> loc.order == 0) mappings
+        toLocationMappings = filter (\loc -> loc.order /= 0) mappings
+    when (null fromLocationMapping || null toLocationMappings) $ throwError (InternalError "Entity Mappings Not Found")
+    let fromLoc = head fromLocationMapping
+    fl <- QL.findById fromLoc.locationId >>= fromMaybeM (InternalError $ "FromLocation not found in booking for fromLocationId: " <> fromLoc.locationId.getId)
+    let toLocMap = last toLocationMappings
```

Comment:
Since in future we will require list of toLocationMappings so I guess we can go with last or mb we can sort on the basis of order in findById so that every time mappings are return in proper order